src/Nethermind/Nethermind.JsonRpc/Data/ReceiptForRpc.cs

```diff
@@ -29,7 +29,7 @@ public ReceiptForRpc(Hash256 txHash, TxReceipt receipt, TxGasInfo gasInfo, int l
             BlobGasPrice = gasInfo.BlobGasPrice;
             From = receipt.Sender;
             To = receipt.Recipient;
-            ContractAddress = receipt.ContractAddress;
+            ContractAddress = receipt.ContractAddress ?? Address.Zero;
```

Comment:
Why would we serialize `Address.Zero` for transactions that don't create contracts?

This will also affect all the RPC method, are we sure we want to do it?

If we don't want to change RPC methods, but we want to change it here, you can inherit `ReceiptForRpc` and add this logic in the derived class.